jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py

```diff
@@ -0,0 +1,128 @@
+from bqplot import pyplot as bqplt
+from ipywidgets import widget_serialization
+from traitlets import Any, Bool, List, Unicode, observe
+
+from jdaviz.configs.imviz.helper import get_top_layer_index
+from jdaviz.core.events import ViewerAddedMessage, ViewerRemovedMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin
+
+__all__ = ['LineProfileXY']
+
+
+@tray_registry('imviz-line-profile-xy', label="Imviz Line Profiles (XY)")
+class LineProfileXY(PluginTemplateMixin):
+    template_file = __file__, "line_profile_xy.vue"
+    viewer_items = List([]).tag(sync=True)
+    selected_viewer = Unicode("").tag(sync=True)
+    plot_available = Bool(False).tag(sync=True)
+    selected_x = Any('').tag(sync=True)
+    selected_y = Any('').tag(sync=True)
+    line_plot_across_x = Any('').tag(sync=True, **widget_serialization)
+    line_plot_across_y = Any('').tag(sync=True, **widget_serialization)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._default_viewer = f'{self.app.config}-0'
+
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._on_viewers_changed)
+        self.hub.subscribe(self, ViewerRemovedMessage, handler=self._on_viewers_changed)
+
+        self._on_viewers_changed()  # Populate it on start-up
+
+    def _on_viewers_changed(self, msg=None):
+        self.viewer_items = self.app.get_viewer_ids()
+
+        # Selected viewer was removed but Imviz always has a default viewer to fall back on.
+        if self.selected_viewer not in self.viewer_items:
+            self.selected_viewer = self._default_viewer
+
+    def reset_results(self):
+        self.plot_available = False
+        self.line_plot_across_x = ''
+        self.line_plot_across_y = ''
+        bqplt.clear()
+
+    # This is also triggered from viewer code.
+    @observe("selected_viewer")
+    def vue_draw_plot(self, *args, **kwargs):
+        """Draw line profile plots for given Data across given X and Y indices (0-indexed)."""
+        if not self.selected_x or not self.selected_y or not self.plugin_opened:
+            return
+
+        viewer = self.app.get_viewer_by_id(self.selected_viewer)
+        i = get_top_layer_index(viewer)
+        data = viewer.state.layers[i].layer
+
+        if isinstance(self.selected_x, str):
+            x = int(round(float(self.selected_x)))
+        else:
+            x = int(round(self.selected_x))
+
+        if isinstance(self.selected_y, str):
+            y = int(round(float(self.selected_y)))
+        else:
+            y = int(round(self.selected_y))
+
+        nx = data.shape[1]
+        ny = data.shape[0]
+        if x < 0 or y < 0 or x >= nx or y >= ny:
+            self.reset_results()
+            return
+
+        x_min, y_min, x_max, y_max = viewer._get_zoom_limits(data)

```

Comment:
I dunno, actually. When someone draw line plots, do they want the plots to stay around until they manually trigger redraw or not? Maybe @camipacifici or @orifox can answer this question.

Reply:
I thought about this more. I don't think it is desirable to have it auto-adjust when zoom changes because that makes it impossible for me to draw line plots, then zoom around for fun, and then do another one. Let's just keep it as-is for now.